docs/topics/multiplatform-mobile/multiplatform-mobile-integrate-in-existing-app.md

```diff
@@ -122,25 +138,26 @@ You can now extract the business logic code to the Kotlin Multiplatform shared m
 
    ![Warnings about platform-dependent code](warnings-android-specific-code.png){width=450}
 
-4. Remove Android-specific code by replacing it with cross-platform Kotlin code or connecting to Android-specific APIs using [`expect` and `actual` declarations](multiplatform-connect-to-apis.md). See the following sections for details.
+4. Remove Android-specific code by replacing it with cross-platform Kotlin code or connecting to Android-specific APIs using [`expect` and `actual` declarations](multiplatform-connect-to-apis.md).
+   See the following sections for details:
 
 #### Replace Android-specific code with cross-platform code {initial-collapse-state="collapsed"}
 
-To make your code work well on both Android and iOS, replace all JVM dependencies with Kotlin dependencies wherever possible.
+To make your code work well on both Android and iOS, replace all JVM dependencies with Kotlin dependencies in the moved `data` directory wherever possible.
 
-1.  In the `login()` function of the `LoginDataSource` class, replace `IOException`, which is not available in Kotlin, with `RuntimeException`.
+1.  In the `LoginDataSource` class, replace `IOException` in the `login()` function, which is not available in Kotlin, with `RuntimeException`.
 
     ```kotlin
     // Before
     return Result.Error(IOException("Error logging in", e))
     ```
 
     ```kotlin
-    //After
+    // After
     return Result.Error(RuntimeException("Error logging in", e))
     ```
 
-2. For email validation, replace the `Patterns` class from the `android.utils` package with a Kotlin regular expression matching the pattern in the `LoginDataValidator` class:
+2. In the `LoginDataValidator` class, for email validation, replace the `Patterns` class from the `android.utils` package with a Kotlin regular expression matching the pattern:
```

Comment:
Moving **For email validation** in the middle of the sentence make it hard to read.